lab_3_generate_by_ngrams/main.py

```diff
@@ -41,6 +44,23 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str):
+            return None
+        self.text = text
+        text = text.lower()
+        split_text = text.split()
+        list_of_text = []
+        for word in split_text:
+            word = list(''.join(letter for letter in word if letter.isalpha()))
+            if not word:
+                continue
+            list_of_text += word
+            list_of_text.append(self._end_of_word_token)
+        if list_of_text.count(self._end_of_word_token) == len(list_of_text):
```

Comment:
Is it actually possible given your code?

Reply:
As I understand if not word we add to the list_of_text only self._end_of_word_token. And it passes one more test with this string